freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/ClusterProxyService.java

```diff
@@ -183,4 +227,12 @@ private ClientCertificate clientCertificates(Stack stack) {
         return clientCertificate;
     }
 
+    private String getEndpointForRegistration(Stack stack, GatewayConfig gatewayConfig, boolean usePrivateIpToTls) {
+        String ipAddresss = gatewayConfig.getPublicAddress();
+        if (usePrivateIpToTls) {
+            ipAddresss = gatewayConfig.getPrivateAddress();
+        }
+        return String.format("https://%s:%d", ipAddresss, stack.getGatewayport());
```

Comment:
protocol should come from `GatewayConfig` as it is used when we are building `apiAddress`

Reply:
The gateway config is updated to use the protocol of cluster proxy. This needs to be the NGINX endpoint protocol not the protocol of cluster proxy. When registering an endpoint in cluster proxy the needs to be the final NGINX endpoint. I will rename the method to make this a little more clear.